cli/commands/root/root.go

```diff
@@ -31,7 +30,7 @@ func Command() *cobra.Command {
 	cmd.PersistentFlags().Bool("insecure-skip-tls-verify", false, "skip TLS certificate verification (not recommended!)")
 	cmd.PersistentFlags().String("config-dir", path.UserConfigDir("sensuctl"), "path to directory containing configuration files")
 	cmd.PersistentFlags().String("cache-dir", path.UserCacheDir("sensuctl"), "path to directory containing cache & temporary files")
-	cmd.PersistentFlags().String("namespace", config.DefaultNamespace, "namespace in which we perform actions")
+	cmd.PersistentFlags().String("namespace", "", "namespace in which we perform actions")
```

Comment:
I'm not sure this is necessary: if namespace is equal to `""`, [it is changed to `config.DefaultNamespace` anyway](https://github.com/sensu/sensu-go/blob/bugfix/resource-delete/cli/client/config/basic/reader.go#L31-L33).

And in a sense, `config.DefaultNamespace` here correctly shows that should the namespace never be explicitly given anywhere (config file or command line), it will default to `config.DefaultNamespace`.

What do you think? Maybe I'm missing something?

Reply:
This is a global and persistent flag, so the default value will populate the namespace field wherever the flag applies. The user in this case is in a non-default namespace, and wants to delete a resource in that namespace without specifying the `--namespace` flag. In order for that to happen, this value cannot have a default. The default should assume the current namespace.

ex. in namespace `devops`
```
$ sensuctl check delete check-in-devops
$ Error: not found
```

Reply:
Alright, I've had to work around some other changes after your global namespace PR. It requires another refactor here, but I think it'll get us the solution we're looking for.